lib/multierror/multierror.go

```diff
@@ -46,41 +57,36 @@ func NewOr(errs []error, fallback error) error {
 // Unwrap for MultiError always returns nil, as there is no reasonable way to implement it.
 //
 // Use As and Is methods, or loop over the list directly to access the underlying errors.
-func (me MultiError) Unwrap() error {
-	return nil
+func (multi MultiError) Unwrap() error {
+	if len(multi) == 0 {
+		return nil
+	}
+	return multi[1:]
 }
 
 // As for MultiError returns the first error that can be considered As the specified target.
-func (me MultiError) As(target interface{}) bool {
-	for _, err := range me {
-		if errors.As(err, target) {
-			return true
-		}
+func (multi MultiError) As(target interface{}) bool {
+	if len(multi) == 0 {
+		return false
 	}
-	return false
+	return errors.As(multi[0], target)
```

Comment:
this does not seem right? eg, `As(` should work if any of the errors in the array can be a target?

Reply:
so here is the source code for errors.Is and As 
https://cs.opensource.google/go/go/+/refs/tags/go1.17.4:src/errors/wrap.go;l=90

The Unwrap() function is supposed to be the equivalent of a list.pop(), which is why all of the tests still pass with these changes.

Right now, it essentially calls errors.As recursively, and for {} loops all get executed in a single pass recursively. 

These changes make it so that it behaves as expected. (I updated this PR's description to include this)